client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClientImpl.java

```diff
@@ -486,4 +507,40 @@ public void close() throws Exception {
             client.close();
         }
     }
+
+    private SSLContext getSSLContext(SchemaRegistryClientConfig config) {
+        try {
+            KeyStore trustStore;
+            if (config.isSystemPropTls()) {
+                return SSLContext.getDefault();
+            } else if (config.isCertificateTrustStore()) {
```

Comment:
Why not just depend on the truststore type instead?

Reply:
we need some variable to know if trust store is certificate or one of java key stores. i have removed the boolean and replaced it with certificate store vs keystore as two separate fields in config.

Reply:
I'd say constraining the supported truststore types to a fixed set and using the corresponding types might make things more explicit here - both in terms of making it explicit in the code what store types are supported as well as this condition.

While Java supports several truststore types, JKS and PKCS12 may be the ones that we might want to support. Keychainstore (for OSX Keychains), Windows-MY (to use Windows store directly), BKS (typically used in Android, etc.) are all specialized ones. Restricting the choices will reduce the scope of testing among other things.

Reply:
For certificates, i agree that we want to restrict it to PEM or DER formats as that is what we expect and parse. The configuration's javadoc calls it out explicitly that certificate should be in one of the two formats. 

but for other store types, since java is able to load the store types or would respond with appropriate exception otherwise, IMO we should not restrict store types to one or two types alone. Since we are not doing anything other than passing the user supplied store (which could also have been set via system properties) so i disagree that we should be restricting it as it will not be absolute restriction.